compiler/resolution/visibleFunctions.cpp

```diff
@@ -516,27 +523,127 @@ static void getVisibleMethodsFirstVisit(const char* name, CallExpr* call,
   }
 }
 
+static void getVisibleMethodsFirstVisitFiltered(const char* name,
+                                                CallExpr* call,
+                                                BlockStmt* block,
+                                                VisibilityInfo* visInfo,
+                                                std::set<BlockStmt*>& visited,
+                                                Vec<FnSymbol*>& visibleFns,
+                                                std::set<const char*> typeNames)
+{
+  // Why does the following statement apply to all blocks,
+  // and not just module or function blocks?
+  //
+  // e.g. in associative.chpl primer, instantiation occurs in a
+  // block that isn't a fn or module block.
+  visited.insert(block);
+  if (visInfo != NULL)
+    visInfo->visitedScopes.push_back(block);
+
+  if (VisibleFunctionBlock* vfb = visibleFunctionMap.get(block)) {
+    // the block defines functions
+
+    if (Vec<FnSymbol*>* fns = vfb->visibleFunctions.get(name)) {
+      forv_Vec(FnSymbol, fn, *fns) {
+        // When private methods and fields are supported, we'll need to extend
+        // this
+        if (fn->isMethod()) {
+          INT_ASSERT(typeNames.size() > 0);
+          std::string nameToCheck = fn->_this->type->symbol->name;
```

Comment:
no need for `std::string` - these are all `astr`s

Reply:
Lol, I  guess I got real nervous and added astr everywhere 😅